US-61

cghende1 opened the pull request:
Testing Procedure:
1. Create a new project and verify that the new project dialog allows you to enter working hours 0-24 for each day of the week.
2. Go to your memoranda save data directory (on Windows it is C:\Users*YourUsername*.memoranda), open the .projects file with 
   your text editor of choice, and verify that your newly-created project has a workinghrs property with the corresponding values
   you entered for the weekly working hours.
3. Set the new project you created (the one for which you entered working hours) as the current project, then choose File > Pack project… 
   and save the project as a .zip file.  Next, delete the project from Memoranda, then re-import it using File > Unpack project…  Finally, 
   open the .projects file in the .memoranda directory and verify that the project has been properly re-imported with the same values for 
   workinghrs that you had entered previously.

WARNING: This new code assumes that all new projects will have been created with WorkingHours, attempting to pack or unpack old projects 
         created without WorkingHours will cause problems!  If need be you can delete your .memoranda directory and then Memoranda will 
         recreate it the next time it is started, thus ensuring that all projects will be created with WorkingHours.

cmbartee committed:
Changed header to read "New Project"

...instead of just "Project"

cmbartee committed:
Changed header to read "New Project"

...instead of just "Project"

cmbartee committed:
Added Day-of-the-Week Selection to New Project

cmbartee committed:
Added Day-of-the-Week Selection to New Project

cmbartee committed:
Polishing presentation a bit

Day-of-Week labels + ComboBoxes alignment/spacing improved

cmbartee committed:
Polishing presentation a bit

Day-of-Week labels + ComboBoxes alignment/spacing improved

cmbartee committed:
Starting on task #63 - added WorkingHours class

Added a class to store the user's working hours per day of the week.

cmbartee committed:
Starting on task #63 - added WorkingHours class

Added a class to store the user's working hours per day of the week.

cmbartee committed:
Working Hours now seems to save to disk properly

Weekly working hours entered in the "New Project" window are now saved
to disk in the .projects file by Memoranda.

-Added get and set for WorkingHours in Project classes
-Changed createProject methods to include WorkingHours
-New Project dialog now saves the entered working hours
-Method added to Util to parse WorkingHours from a string
-Additional constructors added to WorkingHours (default and from
string), as well as a toString method (needed in ProjectImpl.java
WorkingHours set method)

WARNING: Code changed in ProductPackager.java is just a placeholder so
the program will compile; should be changed to actually read in the
saved working hours!

cmbartee committed:
Working Hours now seems to save to disk properly

Weekly working hours entered in the "New Project" window are now saved
to disk in the .projects file by Memoranda.

-Added get and set for WorkingHours in Project classes
-Changed createProject methods to include WorkingHours
-New Project dialog now saves the entered working hours
-Method added to Util to parse WorkingHours from a string
-Additional constructors added to WorkingHours (default and from
string), as well as a toString method (needed in ProjectImpl.java
WorkingHours set method)

WARNING: Code changed in ProductPackager.java is just a placeholder so
the program will compile; should be changed to actually read in the
saved working hours!

cmbartee committed:
WorkingHours are now saved with project pack

Using File > Pack project... to export a project to a .zip file and File
> Unpack project... to import a project from a .zip file now properly
exports and imports the WorkingHours data.

- Also, changed the delimiter to a space instead of a dash, as I was
concerned that the parser would interpret the dash as a negative sign.

NOTE: There was an odd error that I was getting sometimes when trying to
pack a project; it was an array index out of bounds error in
ProductPackager's PackDirectory method:
"java.lang.ArrayIndexOutOfBoundsException: 0
at net.sf.memoranda.util.ProjectPackager.PackDirectory(Unknown Source)"
Upon repeated testing, I could not duplicate the error and packing and
unpacking worked fine, so for now I am doing no further on this issue.

cmbartee committed:
WorkingHours are now saved with project pack

Using File > Pack project... to export a project to a .zip file and File
> Unpack project... to import a project from a .zip file now properly
exports and imports the WorkingHours data.

- Also, changed the delimiter to a space instead of a dash, as I was
concerned that the parser would interpret the dash as a negative sign.

NOTE: There was an odd error that I was getting sometimes when trying to
pack a project; it was an array index out of bounds error in
ProductPackager's PackDirectory method:
"java.lang.ArrayIndexOutOfBoundsException: 0
at net.sf.memoranda.util.ProjectPackager.PackDirectory(Unknown Source)"
Upon repeated testing, I could not duplicate the error and packing and
unpacking worked fine, so for now I am doing no further on this issue.

cghende1 commented:
There have been a lot of commits to this PR. Is it ready for review or are you still working on it?

jgerszty commented:
I get [this](https://gist.github.com/jgerszty/730a8a5b33eed881e85c) error when I try to pack the project without restarting. Works fine after shutting down the program.

cmbartee committed:
Merge branch 'us-61' of https://github.com/cst316/spring16project-Dart into us-61

christopher-henderson commented:
Jacob, has your error been resolved? Can we merge this?

jgerszty commented:
I'm still getting the error if I don't restart Memoranda before trying to pack.

Commit:
Added getter and setter signitures for newly added elapsedTime attribute for a Task.

Commit:
Implemented newly added getter and setter methods for elapsedTime attribute.

Commit:
Updated createTask method to include the elapsedTime attribute.

Commit:
Added elapsedTime column as the second to last column

Commit:
Moved Elapsed Time to be the second column. Adjusted the width

Commit:
Moved Elapsed Time to be the second column. Used getter method to display elapsed time for each task

Commit:
Moved Elapsed Time to be the second column. Applied existing formatting used for priority label to the Elapsed Time column.

Commit:
Updated setter for elapsedTime

Commit:
Added swing timer to calculate elapsed time when user selecttimer buttons for atask

Commit:
Temporarily holding string of zero. Will update get method to return a formated string

Commit:
restricted timer options to allow only one running task

Commit:
Calling new printable time to be displyed in JTable

Commit:
Added popup notification if user selects start timer button before selecting a task

Commit:
Addinteger of seconds to printable time elapsed format method

Commit:
Updated printable format

Commit:
Adjusted time elapsed text to be plain text

Commit:
Added pause function for task timers

Commit:
Added new icons for timer buttons

cghende1 commented:
Confirmed on Mac.

cghende1 commented on src/net/sf/memoranda/ProjectImpl.java:
```diff
@@ -97,6 +98,18 @@ else if (today.after(prEnd)) {
+    
+    public WorkingHours getWorkingHours(){
+    	Attribute d = _root.getAttribute("workingHrs");
+        if (d == null) return null;
```
Please move this to two lines with braces.

cghende1 commented on src/net/sf/memoranda/ProjectImpl.java:
```diff
@@ -97,6 +98,18 @@ else if (today.after(prEnd)) {
+    }
+    
+    public void setWorkingHours(WorkingHours hrs){
+    	if (hrs != null)
```
Please use braces.

cghende1 commented on src/net/sf/memoranda/util/Util.java:
```diff
@@ -81,6 +82,16 @@ public static String getDateStamp(CalendarDate date) {
+    	s = s.trim();
+    	Scanner scan = new Scanner(s);
+    	int[] hrs = new int[7];
+    	for(int i = 0; i < 7; i++)
```
Please add in braces.

jgerszty commented:
Looks good, pliggity plus 1.

cghende1 commented on src/net/sf/memoranda/workinghrs/WorkingHours.java:
```diff
@@ -0,0 +1,138 @@
+	
+	//Setters
+	public void setMondayHours(int hours) throws IllegalArgumentException {
+		if(hours >= 0  && hours <= 24)
```
Braces for alllllll of these.

cmbartee committed:
Added enclosing braces on if statements & for loops

It was brought to my attention that not including enclosing braces for
conditional and loop statements was bad practice and error prone, so I
changed this in order to improve the code quality and reduce the
likelihood of future bug injection.

cmbartee commented:
Fixed the issues with the lack of enclosing braces that cghende1 pointed out.

Commit:
Added condition to add new addtribute for old task objects

Commit:
Uncommented TaskVersioning. Removed statement that assogned a doctype for tasklist file (was throwing a MalformedURL exception)

Commit:
Added message prompt when update or stop button is clicked before selecting a task

Commit:
Merge branch 'master' of https://github.com/cst316/spring16project-Dart into US-13

Commit:
Added check to ensure a .tasklist exists before updating

cmbartee committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into us-61

cmbartee committed:
Fix errors relating to new createProject method

cmbartee committed:
Fixed issue with auto-aggregation

I believe this issue was present in master, and when I merged master's
commits (in order to fix a Travis-CI failure) I found this issue was now
present in this branch.  The issue, specifically, was that a null
pointer exception was being thrown from TaskListImpl.CreateTask() during
the execution of the TaskListTest JUnit test.  The reason why is not
entirely clear to me, however I replaced some code I thought was a bit
dodgy with code that I think is a bit clearer (which fixed the issue) -
so I don't think it's an problem that we don't know where the null
pointer issue was coming from as long as the new code works and appears
correct.  Let me know if you see a problem with the new code.

cghende1 committed:
Merge pull request #19 from cst316/US-13

Us 13

cghende1 committed:
working with merge conflict

cghende1 committed:
working NPE for rebase

cghende1 commented:
Looks good now after an olympic rebase.

crdiaz3 commented:
+1

jgerszty commented:
:+1: :100:

ajcallos commented:
Acceptance test passes
